packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -859,22 +913,26 @@ class StackViewport extends Viewport implements IStackViewport {
     this.setInvertColorGPU(invert);
   }
 
-  private setRotationCPU(rotationCache: number, rotation: number): void {
+  private setRotationCPU(rotation: number): void {
     const { viewport } = this._cpuFallbackEnabledElement;
 
     viewport.rotation = rotation;
-    this.rotationCache = rotation;
-    this.rotation = rotation;
   }
 
-  private setRotationGPU(rotationCache: number, rotation: number): void {
+  private setRotationGPU(rotation: number): void {
+    const { flipVertical } = this.getCamera();
+
     // Moving back to zero rotation, for new scrolled slice rotation is 0 after camera reset
-    this.getVtkActiveCamera().roll(rotationCache);
+    const initialViewUp = flipVertical
```

Comment:
A nice way to undo the rotation/roll without having to know the existing roll/rotation.